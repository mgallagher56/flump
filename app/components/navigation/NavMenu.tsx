import FLPButtonGroup from '~/components/core/buttons/FLPButtonGroup';
import FLPLinkButton from '~/components/core/buttons/FLPLinkButton';
import FLPBox from '~/components/core/structure/FLPBox';

const NavMenu = () => {
  return (
    <FLPBox>
      <FLPButtonGroup gap={4}>
        <FLPLinkButton to="/">Home</FLPLinkButton>
        <FLPLinkButton to="/about">About</FLPLinkButton>
        <FLPLinkButton to="/contact">Contact</FLPLinkButton>
      </FLPButtonGroup>
    </FLPBox>
  );
};

export default NavMenu;
