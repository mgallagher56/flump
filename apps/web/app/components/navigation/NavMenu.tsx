import type { FC } from 'react';

import { useTranslation } from 'react-i18next';
import FLPButtonGroup from '~/components/core/buttons/FLPButtonGroup';
import FLPLinkButton from '~/components/core/buttons/FLPLinkButton';
import FLPBox from '~/components/core/structure/FLPBox';

interface NavMenuProps {
  routes?: { key: string; route: string }[];
}

const NavMenu: FC<NavMenuProps> = ({ routes }) => {
  const { t } = useTranslation();

  return (
    <FLPBox>
      <FLPButtonGroup gap={4}>
        {routes?.map((route) => <FLPLinkButton key={route.key} text={t(route.key)} to={route.route} />)}
      </FLPButtonGroup>
    </FLPBox>
  );
};

export default NavMenu;
