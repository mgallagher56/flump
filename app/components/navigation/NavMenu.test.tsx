import { render } from '@testing-library/react';
import NavMenu from '~/components/navigation/NavMenu';

const mockRoutes = [
  {
    key: 'Home',
    route: '/'
  },
  {
    key: 'Accounts',
    route: '/accounts'
  }
];

describe('<NavMenu />', () => {
  test('renders as expected when not logged in', () => {
    const { baseElement } = render(<NavMenu routes={mockRoutes} />);
    expect(baseElement).toMatchSnapshot();
  });
  test('renders as expected when logged in and showAppLink is true', () => {
    const { baseElement } = render(<NavMenu routes={mockRoutes} />);
    expect(baseElement).toMatchSnapshot();
  });
});
