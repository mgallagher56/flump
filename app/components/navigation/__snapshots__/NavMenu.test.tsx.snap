// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<NavMenu /> > renders as expected when logged in and showAppLink is true 1`] = `
<body>
  <div>
    <div
      class="css-0"
    >
      <div
        class="chakra-button__group css-qul7w4"
        data-orientation="horizontal"
        role="group"
      >
        Home
        Accounts
      </div>
    </div>
  </div>
</body>
`;

exports[`<NavMenu /> > renders as expected when not logged in 1`] = `
<body>
  <div>
    <div
      class="css-0"
    >
      <div
        class="chakra-button__group css-qul7w4"
        data-orientation="horizontal"
        role="group"
      >
        Home
        Accounts
      </div>
    </div>
  </div>
</body>
`;
